packages/visibility_detector/lib/src/render_visibility_detector.dart

```diff
@@ -174,39 +174,54 @@ mixin RenderVisibilityDetectorBase on RenderObject {
     }
     final transform = Matrix4.identity();
 
-    // Create a list of RenderObjects from this to the root, excluding the root
+    // Check if any ancestors decided to skip painting this RenderObject.
+    if (parent != null) {
+      RenderObject ancestor = parent! as RenderObject;
+      RenderObject child = this;
+      while (ancestor.parent != null) {
+        if (!ancestor.paintsChild(child)) {
+          return VisibilityInfo(key: key, size: bounds.size);
+        }
+        child = ancestor;
+        ancestor = ancestor.parent! as RenderObject;
+      }
+    }
+
+    if (layer == null) {
+      return VisibilityInfo(key: key, size: bounds.size);
+    }
```

Comment:
What case is this covering? And why is it returning different data then the "layer is detached and thus invisible" case above?

Reply:
Talked offline, this was added defensively in the wrong way - updated it.